packages/sdk/src/oracle/services/sortedRequests.ts

```diff
@@ -0,0 +1,16 @@
+import { SortedJsMap } from "../../stores";
+import { RequestIndex, RequestIndexes } from "../types/state";
+
+export class SortedRequests {
+  private table = SortedJsMap<string, RequestIndex>();
+  async set(key: string, value: RequestIndex): Promise<void> {
+    await this.table.set(key, value);
+  }
+  async descending(): Promise<RequestIndexes> {
+    // sadly you cannot control lodash sorting descending, so reverse is necessary
+    return (await this.table.values()).reverse();
```

Comment:
Does values return a copy? I only ask because this reverse has side effects on the underlying array

Reply:
yea great point, i didnt even think of that. it does return a copy fortunately!